mqttclients/c/mosquitto_client_extensions/json_handlers/geo_json_handler.c

```diff
@@ -0,0 +1,145 @@
+  {                                                   \
+    if ((x) == NULL)                                  \
+    {                                                 \
+      printf("JSON Parsing Error: %s is NULL\n", #x); \
+      if (jobj_to_free != NULL)                       \
+      {                                               \
+        json_object_put(jobj_to_free);                \
+      }                                               \
+      return -1;                                      \
+    }                                                 \
+  } while (0)
+
+#define RETURN_IF_NON_ZERO(x)                 \
+  do                                          \
+  {                                           \
+    if ((x) != 0)                             \
+    {                                         \
+      printf("JSON Parsing Error: %s\n", #x); \
+      json_object_put(jobj);                  \
+      return -1;                              \
+    }                                         \
+  } while (0)
+
+#define RETURN_IF_NAN(x)                                      \
+  do                                                          \
+  {                                                           \
+    errno = 0;                                                \
+    x;                                                        \
+    if (errno == EINVAL)                                      \
+    {                                                         \
+      printf("JSON Parsing Error: %s is not a number\n", #x); \
+      json_object_put(jobj);                                  \
+      return -1;                                              \
+    }                                                         \
+  } while (0)
+
+geojson_point geojson_point_init()
+{
+  return (geojson_point){ .type = "Point", .coordinates = (geojson_coordinates){ .x = 0, .y = 0 } };
+}
+
+mosquitto_payload mosquitto_payload_init(int max_payload_length)
+{
+  return (mosquitto_payload){ .payload = calloc(1, max_payload_length),
+                              .payload_length = 0,
+                              .max_payload_length = max_payload_length };
+}
+
+void mosquitto_payload_destroy(mosquitto_payload* payload)
+{
+  if (payload->payload != NULL)
+  {
+    free(payload->payload);
+    payload->payload = NULL;
+  }
+  payload->payload_length = 0;
+}
+
+void geojson_point_set_coordinates(geojson_point* pt, double x, double y)
+{
+  pt->coordinates.x = x;
+  pt->coordinates.y = y;
+}
+
+int mosquitto_payload_to_geojson_point(
+    const struct mosquitto_message* message,
+    geojson_point* output)
+{
+  RETURN_IF_NULL(message, NULL);
+  RETURN_IF_NULL(output, NULL);
+
+  json_object* type;
+  json_object* coordinates;
+  json_object* jobj = json_tokener_parse(message->payload);
+
+  RETURN_IF_NULL(type = json_object_object_get(jobj, "type"), jobj);
+  RETURN_IF_NULL(output->type = (char*)json_object_get_string(type), jobj);
```

Comment:
should output be set if this func fails?

Reply:
Since we return an error, it can be assumed that output shouldn't be used/reliable - I'll add a comment indicating that

Reply:
The standard design is to not change any input parameters if there is any error.
That is to avoid creating more memory churn in case there is any memory issue already, making it harder to debug.

Reply:
That should be followed as much as possible as good practice. 
Yes, the code gets longer, but the benefits in case of failure are worth it.

Reply:
Same comment applies to the other functions here.

Reply:
Thanks for the feedback! I've updated this - now the type field is calloc'd in the init function - let me know if there's a way to use this pattern and avoid that!